src/dbnode/storage/database.go

```diff
@@ -694,17 +695,23 @@ func (d *db) writeBatch(
 			errHandler.HandleError(write.OriginalIndex, err)
 		}
 
-		// Need to set the outcome in the success case so the commitlog gets the updated
-		// series object which contains identifiers (like the series ID) whose lifecycle
-		// live longer than the span of this request, making them safe for use by the async
-		// commitlog. Need to set the outcome in the error case so that the commitlog knows
-		// to skip this entry.
-		writes.SetOutcome(i, series, err)
+		if !wasWritten {
```

Comment:
I think you still need to set the outcome I.E dont use an else statement otherwise we'll never report errors back to the caller.

Any way we can improve this with a unit test too?

Reply:
like if err != nil then wasWritten will be false, but we won't record the error in the outcome this way

Reply:
Ah good catch. I'll make a change that if it errors, it will always `SetOutcome(i, _, err)` first